### PR TITLE
Collection code validation

### PR DESCRIPTION
**Pull request by hectorcorrea**
Added database validation to prevent inserting a collection with a duplicates by code, this is particularly important since we have (at least for now) only system collections (RD, LIB, ETD, PPPL).

I had to update several of the tests to prevent them from creating duplication collection codes when running them.

Related to #23

### CODE REVIEW

File: spec/system/user_edit_spec.rb [hunk 1]
```diff
@@ -20,7 +20,10 @@
   end
 
   describe "Non-admin users cannot edit others people data" do
-    before { sign_in user }
+    before do
+      Collection.create_defaults
+      sign_in user
+    end
```
Comment (hectorcorrea) [inline spec/system/user_edit_spec.rb:26]: I can revert this. This is not really need it.

**Review by bess — APPROVED** [review]

File: spec/system/user_edit_spec.rb (hunk 1 above)
Comment (hectorcorrea) [reply spec/system/user_edit_spec.rb]: done

File: app/models/collection.rb [hunk 2]
```diff
@@ -56,4 +56,9 @@ def self.default_for_department(department_number)
       default
     end
   end
+
+  def self.research_data
+    create_defaults
+    Collection.where(code: "RD").first
```
Comment (jrgriffiniii) [inline app/models/collection.rb:62]: 👍

File: db/migrate/20220429133012_add_code_constrain_to_collection.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddCodeConstrainToCollection < ActiveRecord::Migration[6.1]
+  def change
+    add_index :collections, :code, unique: true
```
Comment (jrgriffiniii) [inline db/migrate/20220429133012_add_code_constrain_to_collection.rb:3]: 👍

**Review by jrgriffiniii — APPROVED** [review]

**Comment by jrgriffiniii** [comment]
Please note that this required some updates for the database records within the `staging` environment:

```bash
pdc_describe_staging=> select id,title,code from collections;
 id |                title                | code
----+-------------------------------------+------
  1 | Research Data                       | RD
  3 | Electronic Theses and Dissertations | ETD
  4 | Library Resources                   | LIB
  2 | Princeton Plasma Physics Laboratory | PPPL
  5 | Research Data                       | RD
  6 | Research Data                       | RD
(6 rows)

pdc_describe_staging=> update collections set code='FOO' where id=6;
UPDATE 1
pdc_describe_staging=> update collections set code='BAR' where id=5;
UPDATE 1
pdc_describe_staging=> select id,title,code from collections;
 id |                title                | code
----+-------------------------------------+------
  1 | Research Data                       | RD
  3 | Electronic Theses and Dissertations | ETD
  4 | Library Resources                   | LIB
  2 | Princeton Plasma Physics Laboratory | PPPL
  6 | Research Data                       | FOO
  5 | Research Data                       | BAR
(6 rows)
```

Also note that this shall *not* be necessary for the production environment:

```bash
pdc_describe_prod=> select id,title,code from collections;
 id |                title                | code
----+-------------------------------------+------
  1 | Research Data                       | RD
  2 | Princeton Plasma Physics Laboratory | PPPL
  3 | Electronic Theses and Dissertations | ETD
  4 | Library Resources                   | LIB
(4 rows)
```

**Comment by hectorcorrea** [comment]
@jrgriffiniii thanks for the heads up. I went ahead and linked the one Work that was pointing to the "FOO" collection to point to "RD" and deleted both "FOO" and "BAR" from the staging database.

**Pull request merged**